USWDS-Site - Changelog: Add entry for Button icon alignment

mejiaj opened the pull request:
# Summary

Changelog entry for uswds/uswds#5359.

## Related PR

Related to uswds/uswds#5398.

## Preview link

- [Component entry](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/jm-enhancement-button-icons/components/button/#latest-updates)
- Settings page
  - [Entry](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/jm-enhancement-button-icons/documentation/settings/#component-settings:~:text=Icon%20gap,and%20svg%20icon.)
  - [Channgelog](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/jm-enhancement-button-icons/documentation/settings/#component-settings:~:text=NNNN%2DNN%2DNN,uswds%2Dsite%232207)

## Screenshots

<details><summary>Settings page</summary>
<p></p>

  <img width="900" alt="image" src="https://github.com/uswds/uswds-site/assets/3385219/6e0b0d74-721c-4c7f-9563-5da7dfd3a80b">
  <img width="900" alt="image" src="https://github.com/uswds/uswds-site/assets/3385219/75a3c0d0-a97f-4981-9784-ed5f8c6d04a5">

</details> 


<details><summary>Component page</summary>
<p></p>

  <img width="900" alt="image" src="https://github.com/uswds/uswds-site/assets/3385219/2f457818-776c-4d00-b3b8-0d99c0a5065d">
</details>

amyleadem commented:
FYI - This PR's base branch was set to `release-3.5.1` and was accidentally closed when we reworked the branch name to 3.6.0. I have reopened this PR and have changed the base branch to `main`.

mahoneycm commented on _data/changelogs/component-button.yml:
```diff
@@ -2,6 +2,12 @@ title: Button
+    affectsStyles: true
+    githubPr: 5398
+    githubRepo: uswds
+    versionUswds: 3.5.1
```
Looks like this should be updated to `3.7.0`

amyleadem commented:
Leaving a note that we should include documentation updates for the new button gap setting on both the settings page and the button component page.

amyleadem reviewed (changes requested):
Added some suggestions. We might want to also include a reference to the new setting in the summaryAdditional.

mejiaj commented:
@mahoneycm @amyleadem I've done the following:

- [x] Added GH suggestions.
- [x] Added new entry to settings page.
- [x]  Updated PR description.

mahoneycm reviewed (approved):
Lgtm!

amyleadem commented on _data/changelogs/component-button.yml:
```diff
@@ -2,6 +2,14 @@ title: Button
 changelogURL:
 items:
+  - date: NNNN-NN-NN
+    summary: Improved the vertical alignment of `usa-icon` elements inside of buttons.
```
```suggestion
    summary: Improved the vertical alignment of `usa-icon` elements inside of `usa-button`.
```

amyleadem commented on _data/changelogs/component-button.yml:
```diff
@@ -2,6 +2,14 @@ title: Button
 items:
+  - date: NNNN-NN-NN
+    summary: Improved the vertical alignment of `usa-icon` elements inside of buttons.
+    summaryAdditional: Spacing can be adjusted using the `$theme-button-icon-gap` setting.
```
Wondering if the addition of this setting should be its own separate entry. Either way, I think it would be helpful if we clarified what "spacing" means. Maybe something like this?
```suggestion
    summaryAdditional: Also added the `$theme-button-icon-gap` setting, which adjusts the size of the gap between the button's text and icon.
```

mejiaj replied:
I like the idea of separating the visual from the component setting. I've added a new entry for `$theme-button-icon-gap` [de4cec1].

amyleadem reviewed (changes requested):
Added some suggestions. It would be good to clarify what "spacing"means for `$theme-button-icon-gap`. Also wondering if we should break out the setting addition into its own changelog entry.

mejiaj commented on _data/changelogs/docs-settings.yml:
```diff
@@ -2,9 +2,17 @@ title: Settings
+  - date: NNNN-NN-NN
+    summary: Added button icon gap setting.
+    summaryAdditional: You can now customize the spacing between button text and usa icons with `$theme-button-icon-gap`.
+    affectsGuidance: true
+    affectsSettings: true
+    githubPr: 2207
+    githubRepo: uswds-site
```
Agreed, that makes more sense.

amyleadem reviewed (approved):
LGTM!
